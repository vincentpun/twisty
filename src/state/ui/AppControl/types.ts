export type AppControlSelection = string | AppControlSelectionSection;

export enum AppControlSelectionSection {
  StartingLocation = 'startingLocation',
  DropoffPoints = 'dropoffPoints',
}
