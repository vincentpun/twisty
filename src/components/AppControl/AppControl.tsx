import * as React from 'react';

import HeaderBar from './components/HeaderBar';
import InputFormContainer from './components/InputForm';

interface AppControlProps {
  children?: JSX.Element;
  productName: string;
}

const AppControl = ({ children, productName }: AppControlProps) => (
  <div className="app-control">
    <HeaderBar title={productName} />
    <InputFormContainer />
  </div>
)

export default AppControl;
