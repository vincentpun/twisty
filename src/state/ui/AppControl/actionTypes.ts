export enum AppControlUIActionTypes {
  Select = 'APP_CONTROL_SELECT',

  FetchClientLocationRequest = 'APP_CONTROL_FETCH_CLIENT_LOCATION_REQUEST',
  FetchClientLocationReceive = 'APP_CONTROL_FETCH_CLIENT_LOCATION_RECEIVE',
  FetchClientLocationRequestSuccess = 'APP_CONTROL_FETCH_CLIENT_LOCATION_REQUEST_SUCCESS',
  FetchClientLocationRequestError = 'APP_CONTROL_FETCH_CLIENT_LOCATION_REQUEST_ERROR',

  PickStartingPoint = 'APP_CONTROL_PICK_STARTING_POINT',
  RemoveCoordinates = 'APP_CONTROL_REMOVE_COORDINATES',

  AddDropoffLocation = 'APP_CONTROL_ADD_DROPOFF_LOCATION',
}
