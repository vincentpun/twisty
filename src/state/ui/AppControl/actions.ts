import { createAction, ActionUnion } from 'src/state/actions';
import { AppControlUIActionTypes } from './actionTypes';
import { AppControlSelection, AppControlSelectionSection } from './types';

export const beginPickSession = (section: AppControlSelectionSection) => createAction(
  AppControlUIActionTypes.Select,
  section,
);

export const endPickSession = () => deselect();

export const selectItem = (section: AppControlSelectionSection, id: string) => createAction(
  AppControlUIActionTypes.Select,
  id,
);

export const deselect = () => createAction(
  AppControlUIActionTypes.Select,
  null,
);

export const actions = {
  beginPickSession,
  endPickSession,
  selectItem,
  deselect,
};

export type Actions = ActionUnion<typeof actions>;
