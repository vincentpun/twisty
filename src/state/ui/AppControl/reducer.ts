import { combineReducers } from 'redux';

import { createAction } from 'src/state/actions';
import { AppControlUIActionTypes } from './actionTypes';
import { AppControlSelection } from './types';
import { Actions } from './actions';

export interface AppControlUIState {
  selection: AppControlSelection;
}

const initialState: AppControlUIState = {
  selection: null,
};

export const selection = (state = initialState.selection, action: Actions) => {
  switch (action.type) {
    case AppControlUIActionTypes.Select:
      return action.payload;
    default:
      return state;
  }
}

export default combineReducers<AppControlUIState>({
  selection,
});
