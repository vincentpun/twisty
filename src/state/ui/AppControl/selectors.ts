import { compose, equals, pathOr, values } from 'ramda';

import { State } from 'src/state/reducer';
import { AppControlSelectionSection } from './types';

export const getCurrentSelection = (state: State) => state.ui.appControl.selection;

export const isPickingSection = (state: State, section: AppControlSelectionSection) =>
  compose(
    equals(section),
    pathOr(null, ['ui', 'appControl', 'selection']),
  )(state);

export const isGrabbingClientLocation = (state: State) => state.ui.appControl.isGrabbingClientLocation;

export const hasFailedGrabbingClientLocation = (state: State) => state.ui.appControl.hasFailedGrabbingClientLocation;

export const getStartingCoordinates = (state: State) => {
  const coordinates = state.ui.appControl.startingCoordinates;
  return !!coordinates ? state.ui.appControl.coordinates[coordinates] : null;
};

export const getCoordinates = (state: State) => values(state.ui.appControl.coordinates);

export const getCoordinatesById = (state: State, id: string) => state.ui.appControl.coordinates[id];
