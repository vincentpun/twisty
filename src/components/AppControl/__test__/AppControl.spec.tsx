import * as React from 'react';
import { shallow } from 'enzyme';

import AppControl from '../AppControl';

describe('AppControl', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<AppControl productName="" />);
    expect(wrapper).toMatchSnapshot();
  });
});
