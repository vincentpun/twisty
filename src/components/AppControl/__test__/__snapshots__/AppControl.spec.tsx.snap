// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppControl matches the snapshot 1`] = `
<div
  className="app-control"
>
  <HeaderBar
    title=""
  />
  <InputFormContainer />
</div>
`;
